packages/landing/src/map.ts

```diff
@@ -0,0 +1,107 @@
+    [Epsg.Nztm2000.code]: { lat: -41.277848, lon: 174.6763921, zoom: 2 },
+};
+
+const attributions = '<a href="http://data.linz.govt.nz">LINZ CC BY 4.0 © Imagery Basemap contributors</a>';
+
+export class Basemaps {
+    map: Map;
+    el: HTMLElement;
+    config: MapOptions;
+    location: Location;
+
+    constructor(target: HTMLElement) {
+        this.el = target;
+        this.updateFromUrl();
+    }
+
+    getLayer(): Layer {
+        const projection = this.config.projection;
+
+        if (projection == Epsg.Google) {
+            return new TileLayer({
+                source: new XYZ({ url: WindowUrl.toTileUrl(this.config, MapOptionType.Tile), attributions }),
+            });
+        }
+
+        if (projection == Epsg.Nztm2000) {
+            return new TileLayer({
+                source: new WMTS({
+                    url: WindowUrl.toTileUrl(this.config, MapOptionType.TileWmts),
+                    requestEncoding: 'REST',
+                    attributions: attributions,
+                    projection: projection.toEpsgString(),
+                    tileGrid: NztmOl.TileGrid,
+                    // These keys arent really needed but need to be strings
+                    layer: '',
+                    style: '',
+                    matrixSet: '',
+                }),
+            });
+        }
+
+        throw new Error('Unable to find layer for projection: ' + projection);
+    }
+    /** Import location and rendering information from the url */
+    private updateFromUrl(): void {
+        this.config = WindowUrl.fromUrl(window.location.search);
+        const projection = this.config.projection;
+        const location = { ...DefaultCenter[projection.code], ...WindowUrl.fromHash(window.location.hash) };
+
+        const loc = proj.transform([location.lon, location.lat], UrlProjection, `EPSG:${projection}`);
+        let resolutions: undefined | number[] = undefined;
+        if (projection == Epsg.Nztm2000) {
+            resolutions = NztmOl.resolutions;
+        }
+        const view = new View({ projection: projection.toEpsgString(), center: loc, zoom: location.zoom, resolutions });
+
+        this.map = new Map({
+            target: this.el,
+            view,
+            layers: [this.getLayer()],
+        });
+
+        this.map.addEventListener('postrender', this.postRender);
+    }
+
+    updateUrlTimer: unknown | null = null;
+    postRender = (): void => {
+        if (this.updateUrlTimer != null) return;
+        this.updateUrlTimer = setTimeout(() => this.updateUrl(), 1000);
+    };
+
+    getLocation(): MapLocation {
+        const view = this.map.getView();
+        const center = view.getCenter();
+        if (center == null) throw new Error('Invalid Map location');
+        const [lon, lat] = proj
+            .transform(center, `EPSG:${this.config.projection}`, UrlProjection)
+            .map((c: number) => Math.floor(c * 10e5) / 10e5); // Limit to 5 decimal places (TODO is 5 enough)
```

Comment:
Google Maps uses 7, I believe

Reply:
If it's just for centring the map 5dp (~1m) should be ok. 7dp will give you ~1cm precision.

Reply:
I was meaning to check what google maps did, they use 7 maybe we should use 7 too, simple fix

Reply:
Switched to 7dp